pyrealm/phenology/fapar_limitation.py

```diff
@@ -0,0 +1,274 @@
+
+    return precip_molar
+
+
+class FaparLimitation:
+    r"""FaparLimitation class to compute fAPAR_max and LAI.
+
+    This class takes the annual total potential GPP and precipitation, the annual
+    mean CA, Chi and VPD, as well as the aridity index and some constants to compute
+    the annual peak fractional absorbed photosynthetically active radiation (
+    fAPAR_max) and annual peak Leaf Area Index (LAI).
+    """
+
+    def _check_shapes(self) -> None:
+        """Internal class to check all the input arrays have the same size."""
+
+        assert len(self.annual_total_potential_gpp) == len(self.annual_mean_ca)
+        assert len(self.annual_mean_ca) == len(self.annual_mean_chi)
+        assert len(self.annual_mean_chi) == len(self.annual_mean_vpd)
+        assert len(self.annual_mean_vpd) == len(self.annual_total_precip)
+        if np.shape(self.aridity_index) != ():
+            assert len(self.annual_total_precip) == len(self.aridity_index)
+
+    def __init__(
+        self,
+        annual_total_potential_gpp: NDArray[np.float64],
+        annual_mean_ca: NDArray[np.float64],
+        annual_mean_chi: NDArray[np.float64],
+        annual_mean_vpd: NDArray[np.float64],
+        annual_total_precip: NDArray[np.float64],
+        aridity_index: NDArray[np.float64],
+    ) -> None:
+        r"""Annual peak fractional absorbed photosynthetically active radiation (fAPAR).
+
+        Computes fAPAR_max as minimum of a water-limited and an energy-limited
+        quantity.
+
+        Args:
+            annual_total_potential_gpp: Aka A_0, the annual sum of potential GPP.
+              Potential GPP would be achieved if fAPAR = 1. [mol m^{-2} year^{-1}]
+            annual_mean_ca: Ambient CO2 partial pressure. [Pa]
+            annual_mean_chi: Annual mean ratio of leaf-internal CO2 partial pressure to
+               c_a during the growing season (>0℃). [Pa]
+            annual_mean_vpd: Aka D, annual mean vapour pressure deficit (VPD) during the
+              growing season (>0℃). [Pa]
+            annual_total_precip: Aka P, annual total precipitation. [mol m^{-2}
+             year^{-1}]
+            aridity_index: Aka AI, climatological estimate of local aridity index.
+        """
+
+        self.annual_total_potential_gpp = annual_total_potential_gpp
+        self.annual_mean_ca = annual_mean_ca
+        self.annual_mean_chi = annual_mean_chi
+        self.annual_mean_vpd = annual_mean_vpd
+        self.annual_total_precip = annual_total_precip
+        self.aridity_index = aridity_index
+
+        self._check_shapes()
+
+        self.phenology_const = PhenologyConst()
+
+        #  f_0 is the ratio of annual total transpiration of annual total
+        #  precipitation, which is an empirical function of the climatic Aridity Index
+        #  (AI).
+        a = self.phenology_const.f0_coefficients[0]
+        b = self.phenology_const.f0_coefficients[1]
+        c = self.phenology_const.f0_coefficients[2]
```

Comment:
Python can unpack when assigning:

```suggestion
        a, b, c = self.phenology_const.f0_coefficients
```